### ISSUE TITLE
Mixed capitalisation ending in lowercase (e.g. pluralized acronyms) stops errors in sentence being detected

### ISSUE THREAD

**Issue by HawkinsT** [languagetool-org/languagetool#7414]
If I have a word with multiple capital letters ending in a lowercase letter, languagetool stops checking for some rules in the rest of the sentence. For example, if I have a sentence that's over 40 words and contains the pluralized acronym, SQUIDs, no warning will be given. If I change this word to SQUID, then the long sentence warning will reappear. This is also the case with the Oxford comma rule and seemingly several others.

Example:
`This is a very long sentence about SQUIDs which will not show an error for some reason, even though I'm really not giving the reader the time to breath or collect their thoughts since it goes on and one for over 40 words.`

will not show a warning, but:

`This is a very long sentence about a SQUID which will not show an error for some reason, even though I'm really not giving the reader the time to breath or collect their thoughts since it goes on and one for over 40 words.`

does.

**Comment by danielnaber** [comment]
Thanks for the report. I cannot reproduce this yet, though. Are you sure you have picky mode activated in both cases? Where does this happen, in the add-on or on languagetool.org?

**Comment by HawkinsT** [comment]
Sorry, I've been accessing a local version of languagetool via the ltex lsp. To make sure it wasn't an issue with ltex I also tested this on languagetool.org but it seems I used a bad example with the Oxford comma rule, as it turns out this is never triggered for a list of three items when one is an acronym (I don't know, but perhaps this is desired behaviour). Therefore, I believe it's actually a problem with the lsp and I'll report it there directly. Apologies!
